addons/common/functions/fnc_exportMissionSQF.sqf

```diff
@@ -0,0 +1,428 @@
+#include "script_component.hpp"
+/*
+ * Author: mharis001
+ * Returns SQF code that can be executed to restore the current mission.
+ * Players and dead objects are ignored.
+ *
+ * Arguments:
+ * 0: Position <ARRAY> (default: [0, 0, 0])
+ * 1: Radius <NUMBER> (default: -1)
+ *   - in meters, -1 for entire mission
+ * 2: Include Waypoints <BOOL> (default: true)
+ * 3: Include Markers <BOOL> (default: false)
+ * 4: Curator Editable Objects Only <BOOL> (default: false)
+ *
+ * Return Value:
+ * Mission SQF <STRING>
+ *
+ * Example:
+ * [] call zen_common_fnc_exportMissionSQF
+ *
+ * Public: No
+ */
+
+#define FORMAT_OBJ_VAR(index) format ["_object%1", index]
+#define FORMAT_GRP_VAR(index) format ["_group%1", index]
+
+#define NEWLINE toString [ASCII_NEWLINE]
+
+params [
+    ["_position", [0, 0, 0], [[]], [2, 3]],
+    ["_radius", -1, [0]],
+    ["_includeWaypoints", true, [true]],
+    ["_includeMarkers", false, [true]],
+    ["_editableOnly", false, [true]]
+];
+
+// Keep track of all processed objects and groups, their index in their corresponding array
+// is used to determine the variable name used in the exported SQF
+private _processedObjects = [];
+private _processedGroups = [];
+
+// Separate the exported SQF into different sections, this is used to ensure the correct
+// ordering of the output (for example, applying group properties after all units are created)
+private _outputGroups1 = [];
+private _outputObjects = [];
+private _outputGroups2 = [];
```

Comment:
Maybe some comments to describe why are they named so similarly?

Reply:
L43 explains why there are 2 group arrays.